src/source/src/table.rs

```diff
@@ -94,9 +94,10 @@ impl TableDmlHandle {
                 )));
             }
             let len = guard.changes_txs.len();
+            // used session id instead of txn_id to choose channel so that we can preserve transaction order in the same session.
             let sender = guard
                 .changes_txs
-                .get((txn_id % len as u64) as usize)
+                .get((session_id % len as u32) as usize)
```

Comment:
Use session id instead of txn_id to choose channel so that we can preserve transaction order in the same session